src/rulemanager.py

```diff
@@ -18,6 +20,24 @@
 
 This document specifies the OpenSCHC rule data model, which is based on JSON.
 
+# File Format
+
+When the rules are stored in a file, the file format is JSON as specified here.
+When the rules are given or handled in Python, the data format is the equivalent
+of the parsed JSON (e.g. dictionaries, lists, etc.).
+
+# Internal data structure
+
+The reason for the internal data structure given abose is as follows:
+- in SCHC, a Rule ID is defined only in the context of one device 
+("multiple Dev instances, which refer to different header compression Contexts, MAY reuse the same Rule ID for different Rules", SCHC RFC-TBD-XXX, section 7.2),
+- hence the database of rules, is chosen to be a list of device contexts 
+ (defined by device identifiers), each of them associated with a set of rules.
+The device_context is specified below in section Context, and looks like:
+  {"DeviceID": 0x1234567, "SoR": {<rule 1>, <rule 2>, ...}}
```

Comment:
Since JSON only has arrays, we suggest to use arrays in Python as well instead of sets.
Laurent and Dominique

Reply:
I updated the comment(s), the code was already assuming lists (using "append" etc.)